PWGHF/HFC/TableProducer/femtoDreamProducer.cxx

```diff
@@ -0,0 +1,530 @@
+
+      auto fillTable = [&](int CandFlag,
+                           int FunctionSelection,
+                           float BDTScoreBkg,
+                           float BDTScorePrompt,
+                           float BDTScoreFD) {
+        if (FunctionSelection >= 1){
+        // Fill tracks if it is not filled for Lc Candidate in an event
+            if (!isTrackFilled) {
+                isTrackFilled = fillTracksForCharmHadron<false>(tracks, trackPos1, trackNeg, trackPos2, sizeCand);
+
+                // If track filling was successful, fill the collision table
+                if (isTrackFilled) {
+                    outputCollision(vtxZ, mult, multNtr, spher, mMagField);
+                }
+            }
+
+            // fill collision table if track table is filled, i.e., there is at least one Lc-p pair
+            if (isTrackFilled) {
+                // Row for candidate charm hadron
+                rowCandCharmHad(
+                    outputCollision.lastIndex(),
+                    trackPos1.sign() + trackNeg.sign() + trackPos2.sign(),
+                    trackPos1.globalIndex(),
+                    trackNeg.globalIndex(),
+                    trackPos2.globalIndex(),
+                    trackPos1.pt(),
+                    trackNeg.pt(),
+                    trackPos2.pt(),
+                    trackPos1.eta(),
+                    trackNeg.eta(),
+                    trackPos2.eta(),
+                    trackPos1.phi(),
+                    trackNeg.phi(),
+                    trackPos2.phi(),
+                    1 << CandFlag,
+                    BDTScoreBkg,
+                    BDTScorePrompt,
+                    BDTScoreFD);
+
+                // Row for MC candidate charm hadron (if constexpr isMC)
+                if constexpr (isMC) {
+                    rowCandMCCharmHad(
+                        candidate.flagMcMatchRec(),
+                        candidate.originMcRec());
+                }
+            }
+      } };
+
+      fillTable(0, candidate.isSelLcToPKPi(), outputMlPKPi.at(0), outputMlPKPi.at(1), outputMlPKPi.at(2));
+      fillTable(1, candidate.isSelLcToPiKP(), outputMlPiKP.at(0), outputMlPiKP.at(1), outputMlPiKP.at(2));
+    }
+  }
+
+  template <typename ParticleType>
+  void fillCharmHadMcGen(ParticleType particles)
+  {
+    // Filling particle properties
+    rowCandCharmHadGen.reserve(particles.size());
+    for (const auto& particle : particles) {
+      if (std::abs(particle.flagMcMatchGen()) == 1 << aod::hf_cand_3prong::DecayType::LcToPKPi) {
+        rowCandCharmHadGen(
+          particle.mcCollisionId(),
+          particle.flagMcMatchGen(),
+          particle.originMcGen());
+      }
+    }
+  }
+
+  void processDataCharmHad(FemtoFullCollision const& col, aod::BCsWithTimestamps const&,
+                           FemtoHFTracks const& tracks,
+                           soa::Filtered<CandidateLc> const& candidates)
```

Comment:
One argument per line

Reply:
Hi @vkucera, you put a comment just below this comment regarding "void fillCharmHadMCGen(TrackType cons", here you are referring to old file, I have updated this file yesterday